pandas/core/frame.py

```diff
@@ -3909,43 +3909,59 @@ def shift(self, periods=1, freq=None, axis=0):
     def set_index(self, keys, drop=True, append=False, inplace=False,
                   verify_integrity=False):
         """
+        An index is created with existing columns.
```

Comment:
I find this a bit confusing as the summary line (because it seems to suggest that an index is returned, since that is created). 
I found the previous "Set the DataFrame index using existing columns" a bit clearer. @datapythonista thoughts?